src/main/java/org/craftercms/engine/service/context/SiteContextManager.java

```diff
@@ -303,6 +340,48 @@ public void startContextRebuild(String siteName, boolean fallback, Consumer<Site
         });
     }
 
+    /**
+     * Get a site context and initializing if it does not exist.
+     * If in preview mode, do not retry on failure.
+     * If in live mode, retry on failure. After the max retries count, throw an exception.
+     * @param siteName the site name of the context
+     * @return the site context
+     * @throws InterruptedException if the current thread is interrupted while waiting
+     */
+    protected SiteContext getContextWithInitializing(String siteName) throws InterruptedException {
```

Comment:
Please rename to `getContextWithRetries`

Reply:
Fixed. Thank you!